lib/handler/retry-handler.js

```diff
@@ -296,7 +310,10 @@ class RetryHandler {
   onData (chunk) {
     this.start += chunk.length
 
-    return this.handler.onData(chunk)
+    const written = this.handler.onData(chunk)
```

Comment:
I use the return of `onData` to indicate the data is written to downstream.
When it return `false` from the first write, we still have chance to retry.

Reply:
You can't do that. It only indicated back pressure. Not whether the data was sent or not.

Reply:
Doesn't `false` means the data didn't push to the buffer?
If yes, then it is what we need for this case.

Reply:
Nevermind, I found the answer. It is no way back.